### PR TITLE
Update e2e test images url

### LINKED ISSUE #96770: Use k8s.gcr.io/e2e-test-images instead of gcr.io/kubernetes-e2e-test-images for e2e test images
Part of umbrella to migrate kubernetes e2e test images/registries to community-owned infrastructure: https://github.com/kubernetes/k8s.io/issues/1458

We should migrate away from the google.com-owned gcr.io/kubernetes-e2e-test-images and instead use the community-owned k8s.gcr.io/e2e-test-images repository.

Affected images are those that use `E2eRegistry` or `E2eVolumeRegistry`.  ~We may also want to consider pushing `DockerLibraryRegistry` images into k8s.gcr.io/e2e-test-images~ (moved to https://github.com/kubernetes/kubernetes/issues/97027)

https://github.com/kubernetes/kubernetes/blob/1d4c0ad6f33f2bc42555becdb5d74e570c0730f3/test/utils/image/manifest.go#L203-L245

My guess at the steps required:
- push old images from `gcr.io/kubernetes-e2e-test-images` to `gcr.io/k8s-staging-e2e-test-images`
- promote images used by latest supported patch releases and in-development release
- create postsubmit jobs for each test/image dir (done via https://github.com/kubernetes/kubernetes/issues/96432)
- PR change values of `E2eRegistry` and `E2eVolumeRegistry` to `k8s.gcr.io/e2e-test-images`, remove `PromoterE2eRegistry`
- subsequent trial PRs to run through building/promoting an e2e image

This is release-notes worthy and possibly announcement worthy.

/sig testing
/sig release
/wg k8s-infra
/area test-infra
/area test
/area images-registry
/priority important-soon

FYI @aojea @claudiubelu 
FYI @kubernetes/release-engineering

### PR DESCRIPTION
**Pull request by claudiubelu**
#### What type of PR is this?

/kind cleanup

/sig testing
/priority important-soon

#### What this PR does / why we need it:

Removes any reference from the registry ``gcr.io/kubernetes-e2e-test-images`` in kubernetes/kubernetes, replacing them with ``k8s.gcr.io/kubernetes-e2e-test-images``. In some cases, the images had to be updated since a few things have changed since
their original implementation, most notably being the fact that some of the images have been centralized into the agnhost image.

Co-Authored-By: Claudiu Belu <cbelu@cloudbasesolutions.com>

#### Which issue(s) this PR fixes:

Fixes #96770

#### Special notes for your reviewer:

Cherry-picked from: #100470

#### Does this PR introduce a user-facing change?

```release-note
``gcr.io/kubernetes-e2e-test-images`` will no longer be used in E2E / CI testing, ``k8s.gcr.io/e2e-test-images`` will be used instead.
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by spiffxp** [comment]
@claudiubelu this needs a release note, which I don't have the bandwidth to draft today...

If I knew the answer to this before, it's been paged out of my brain, and I'll need time to page back in... what are the implications for people who are used to mirroring/air-gapping `kubernetes-e2e-test-images`?  Do they need to use a different invocation today? Would things break if they used the same invocation as they're using for v1.21? etc.

/cc @wilsonehusin @dims 
I want eyes from someone who's used to air-gapping these images, if it's not you maybe you can point to someone who would be more appropriate?

**Comment by dims** [comment]
@spiffxp for the air gapped  scenarios that i have seen, folks just throw all the necessary images in the same bucket, it's us who have them in different buckets :)

cc @johnSchnake @vladimirvivien

**Comment by vladimirvivien** [comment]
As far as an airgap use case for, say, Sonobuoy, this should be Ok as long as e2e test binary returns the proper repo when queried .

File: test/e2e/testing-manifests/serviceloadbalancer/netexecrc.yaml
Comment (smarterclayton) [inline test/e2e/testing-manifests/serviceloadbalancer/netexecrc.yaml:15]: Is this file even used anymore?  I couldn't find any references to it (doing a very simple search in the repo), and the interesting part of a lot of these changes is that they aren't used in e2e anymore (but a few are used in the `test/cmd` shell scripts).  The key point is that these *can't* be used inside of e2e binaries because they aren't subject to dynamic replacement (there's no code reading these and then looking up the correct value of the image before running it), so it's possible that we should move these out of test/e2e and into test/cmd/ (where appropriate).

For example:

```
  kubectl create -f test/e2e/testing-manifests/kubectl/agnhost-primary-pod.yaml "${kube_flags[@]}"
```

in `test/cmd/core.sh` is the only reference to the file above, so a separate PR moving these is probably appropriate.

Comment (spiffxp) [reply test/e2e/testing-manifests/serviceloadbalancer/netexecrc.yaml:15]: I could have sworn we had code that dynamically substituted to support repo mappings passed in via `KUBE_TEST_REPO`

https://github.com/kubernetes/kubernetes/blob/9f47110aa29094ed2878cf1d85874cb59214664a/test/utils/image/manifest.go#L253-L255

Maybe it's been removed since I last looked?

Comment (spiffxp) [reply test/e2e/testing-manifests/serviceloadbalancer/netexecrc.yaml:15]: Ah, here it is https://github.com/kubernetes/kubernetes/blob/9f47110aa29094ed2878cf1d85874cb59214664a/test/utils/image/manifest.go#L352-L356

Used by
https://github.com/kubernetes/kubernetes/blob/9f47110aa29094ed2878cf1d85874cb59214664a/test/e2e/storage/utils/create.go#L626-L629

Comment (spiffxp) [reply test/e2e/testing-manifests/serviceloadbalancer/netexecrc.yaml:15]: tl;dr I don't know if you were talking about the things in this specific file or these manifests in general, but they can have their images dynamically substituted

Comment (claudiubelu) [reply test/e2e/testing-manifests/serviceloadbalancer/netexecrc.yaml:15]: So, indeed, some of these files do not seem to be used anymore, like:

```
test/e2e/testing-manifests/ingress/http/rc.yaml
test/e2e/testing-manifests/ingress/http2/rc.yaml
test/e2e/testing-manifests/ingress/multiple-certs/rc.yaml
test/e2e/testing-manifests/ingress/pre-shared-cert/rc.yaml
test/e2e/testing-manifests/ingress/static-ip/rc.yaml
test/e2e/testing-manifests/serviceloadbalancer/netexecrc.yaml
```

Some of them are still being used, like the ``test/e2e/testing-manifests/ingress/neg*`` ones. 

``test/e2e/testing-manifests/kubectl/agnhost-primary-pod.yaml`` seems to be used only in ``test/cmd/core.sh``, so it could probably be moved out of ``test/e2e`` if needed.

And indeed, for some files the used images are dynamically substituted, which can be useful for airgapped environments (originally, AFAIK, the mechanism was introduced so we can use container images with Windows support for Windows test runs, before all the necessary work merged into k/k test images). But any other cleanup / change can be done in another PRs, specifically targeted towards those purposes. This PR just cleans up the gcr.io references.

**Review by spiffxp — APPROVED** [review]
/approve
/lgtm

**Comment by claudiubelu** [comment]
@smarterclayton , it seems you're an approver for sample-apiserver. Is there anything else that needs to be addressed for this PR?

File: staging/src/k8s.io/sample-apiserver/artifacts/example/deployment.yaml
Comment (spiffxp) [inline staging/src/k8s.io/sample-apiserver/artifacts/example/deployment.yaml:24]: /cc @liggitt @lavalamp @deads2k 
we need an /approve from someone in sample-apiserver/OWNERS for this comment change

**Comment by liggitt** [comment]
/approve

for comment change

**Pull request merged**
